assets/app/scripts/services/data.js

```diff
@@ -182,6 +182,37 @@ angular.module('openshiftConsole')
     return deferred.promise;
   };
 
+// type:      API type (e.g. "pods")
+// name:      API name, the unique name for the object
+// object:    API object data(eg. { kind: "Build", parameters: { ... } } )
+// context:   API context (e.g. {project: "..."})
+// opts:      http - options to pass to the inner $http call
+// Returns a promise resolved with response data or rejected with {data:..., status:..., headers:..., config:...} when the delete call completes.
+  DataService.prototype.customCreate = function(type, name, object, context, opts) {
```

Comment:
instead of writing a "customCreate" function, @liggitt  and I discussed that the "type" that is passed in should be "buildConfigs/instantiate" or "build/clone" and then the urlForType needs to understand that a type with a / in it like "build/clone" means the resource type is "build" and the action is "clone" and then build the URL as resource/name/action, this keeps us from needing to create a bunch of custom URL templates when there is really a common pattern, instead you would have something like:

```
var URL_NAMESPACED_OBJECT_ACTION = URL_ROOT_TEMPLATE + "namespaces/{namespace}/{type}/{id}/{action}{?q*}";
```